gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/upgrade/EventsLatestUpgrader.java

```diff
@@ -0,0 +1,157 @@
+        @Lazy DictionaryRepository dictionaryRepository,
+        @Lazy OrganizationRepository organizationRepository,
+        @Lazy EventRepository eventRepository,
+        @Lazy EventLatestRepository eventLatestRepository
+    ) {
+        super(InstallationService.EVENTS_LATEST_UPGRADER_STATUS);
+        this.apiRepository = apiRepository;
+        this.dictionaryRepository = dictionaryRepository;
+        this.organizationRepository = organizationRepository;
+        this.eventRepository = eventRepository;
+        this.eventLatestRepository = eventLatestRepository;
+    }
+
+    @Override
+    public int getOrder() {
+        return 600;
+    }
+
+    @Override
+    protected void processOneShotUpgrade() throws TechnicalException {
+        migrateApiEvents();
+        migrateDictionaryEvents();
+        migrateOrganizationEvents();
+    }
+
+    private void migrateApiEvents() throws TechnicalException {
+        log.info("Starting migrating latest event for APIs");
+        modelCounter = 0;
+        eventsForModelCounter = 0;
+        List<ApiCriteria> emptyCriteria = List.of();
+        Page<String> firstPage =
+            this.apiRepository.searchIds(emptyCriteria, new PageableBuilder().pageNumber(0).pageSize(BULK_SIZE).build(), null);
+        if (firstPage != null && firstPage.getContent() != null) {
+            migrateEvents(Event.EventProperties.API_ID.getValue(), firstPage.getContent());
+
+            // Cover others pages if required
+            if (firstPage.getTotalElements() > firstPage.getPageElements()) {
+                long pageNumber = (int) Math.ceil((double) firstPage.getTotalElements() / BULK_SIZE);
+                for (int pageIdx = 1; pageIdx < pageNumber; pageIdx++) {
+                    Page<String> nextPage =
+                        this.apiRepository.searchIds(
+                                emptyCriteria,
+                                new PageableBuilder().pageNumber(pageIdx).pageSize(BULK_SIZE).build(),
+                                null
+                            );
+                    migrateEvents(Event.EventProperties.API_ID.getValue(), nextPage.getContent());
+                }
+            }
+        }
+        log.info("{} events regarding {} APIs have been migrated", eventsForModelCounter, modelCounter);
+    }
+
+    private void migrateDictionaryEvents() throws TechnicalException {
+        log.info("Starting migrating latest event for dictionaries");
+        modelCounter = 0;
+        eventsForModelCounter = 0;
+        Set<Dictionary> dictionaries = this.dictionaryRepository.findAll();
+        if (dictionaries != null) {
+            List<String> ids = dictionaries.stream().map(Dictionary::getId).collect(Collectors.toList());
+            migrateEvents(Event.EventProperties.DICTIONARY_ID.getValue(), ids);
+        }
+        log.info("{} events regarding {} dictionaries have been migrated", eventsForModelCounter, modelCounter);
+    }
```

Comment:
Latest event is also used on mapi side but et's discuss it with @jhaeyaert because indeed we could have an issue here if latest event is STOP, the gateway won't detect it.